SharpDenizenTools/MetaObjects/MetaTypeData.cs

```diff
@@ -0,0 +1,51 @@
+﻿using FreneticUtilities.FreneticExtensions;
+using SharpDenizenTools.MetaHandlers;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace SharpDenizenTools.MetaObjects
+{
+    /// <summary>A generic-less <see cref="MetaTypeData{T}"/> interface.</summary>
+    public interface IMetaTypeData
+    {
+        /// <summary>Getter to create a new instance of this meta type.</summary>
+        Func<MetaObject> Getter { get; }
+
+        /// <summary>The meta type this data is for.</summary>
+        MetaType Type { get; }
+
+        /// <summary>Creates a new <see cref="MetaObject"/> using <see cref="Getter"/> and sets it's type.</summary>
+        public MetaObject CreateNewMeta()
+        {
+            MetaObject newMeta = Getter();
+            newMeta.Type = Type;
+            return newMeta;
+        }
+
+        /// <summary>Gets a <see cref="MetaObject"/> by name, or <c>null</c> if one by that name doesn't exist.</summary>
+        MetaObject MetaByNameIfPresent(string name);
```

Comment:
Ideally use the `TryGet` pattern, ie `bool TryGetObject(string name, out MetaObject obj)`

Reply:
Changed to that